### PR TITLE
Fix: Macro version was broken (Cabal)

### PR DESCRIPTION
**Pull request by andys8**
Solves https://github.com/haskell-hvr/cabal-plan/issues/87

### CODE REVIEW

File: src-exe/LicenseReport.hs [hunk 1]
```diff
@@ -40,7 +40,7 @@ import           Prelude.Compat
 
 import Cabal.Config (readConfig, Config (..), cfgRepoIndex, hackageHaskellOrg)
 
-#if MIN_VERSION_Cabal(3,2,0)
+#if MIN_VERSION_Cabal_syntax(3,2,0)
```
Comment (andys8) [inline src-exe/LicenseReport.hs:43]: :information_source: Please double check if the actual version bound (3.2.0) still makes sense after switching to `Cabal-syntax`

Comment (phadej) [reply src-exe/LicenseReport.hs:43]: The version guards can be removed, as we always require at least `Cabal-syntax >=3.8.1.0`

**Review by phadej — CHANGES_REQUESTED** [review]
This in turn doesn't work if compiled against `CAbal`

File: src-exe/LicenseReport.hs (hunk 1 above)
Comment (andys8) [reply src-exe/LicenseReport.hs:43]: Right

**Comment by andys8** [comment]
> This in turn doesn't work if compiled against `CAbal`

What does this mean exactly?

I don't know if I fully understand what's going on with that condition. The way I interpreted it, is that only the `if flag(license-report)` part containes a `Cabal/Cabal-syntax` dependency.

https://github.com/andys8/cabal-plan/blob/63b5c003074a3bd5bea4b6e9d7f3d06eefed20fc/cabal-plan.cabal#L129-L134

And therefore `#if defined(MIN_VERSION_Cabal[_syntax])` is only a guard to make sure the flag was activated. Therefore it otherwise results in 

> Please recompile/reinstall `cabal-plan` with the `license-report` Cabal flag activated.

https://github.com/andys8/cabal-plan/blob/63b5c003074a3bd5bea4b6e9d7f3d06eefed20fc/src-exe/LicenseReport.hs#L12


So, how are users installing it against `Cabal`? Is this a compat thing I don't know of? Or if `cabal-plan` is used as library? And most important: How can it be fixed - since I think the current state is broken?

**Comment by phadej** [comment]
Squashed in https://github.com/haskell-hvr/cabal-plan/pull/89, thanks.

**Comment by phadej** [comment]
Release https://hackage.haskell.org/package/cabal-plan-0.7.2.3

**Pull request closed without merging**